src/util/std_expr.h

```diff
@@ -664,8 +664,8 @@ class binary_predicate_exprt:public binary_exprt
 
     DATA_CHECK(
       vm,
-      expr.type().id() == ID_bool,
-      "result of binary predicate expression should be of type bool");
+      expr.type().id() == ID_bool || expr.type().id() == ID_vector,
+      "result of binary predicate expression should be of type bool or vector");
```

Comment:
@kroening I quite dislike that I had to make the above change: this is really to make `--validate-goto-model` pass, which is currently applied before and (!) after calls to `remove_vector`. We should probably discuss this.

Reply:
Yes, not super. One could argue that something that returns a vector isn't a predicate.

Reply:
Perhaps the better solution would be introducing ids along the lines of `vector_equal`, `vector_gt`, etc?

Reply:
Not absolutely necessary -- we do overload many other IDs, but I admit that others may be surprised to see a non-boolean equality. `vector_equal` etc. may then be the best compromise.